File: app/src/main/java/com/example/unsplash/model/datasource/FavouritesDataSourceFactory.java
package com.example.unsplash.model.datasource;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;
import android.arch.paging.PageKeyedDataSource;

import com.example.unsplash.model.models.Photo;

public class FavouritesDataSourceFactory extends DataSource.Factory {

    private MutableLiveData<PageKeyedDataSource<Integer, Photo>> favouritesLiveDataSource = new MutableLiveData<>();

    @Override
    public DataSource create() {
        FavouritesDataSource favouritesDataSource = new FavouritesDataSource();
        favouritesLiveDataSource.postValue(favouritesDataSource);
        return favouritesDataSource;
    }

    public MutableLiveData<PageKeyedDataSource<Integer, Photo>> getFavouritesLiveDataSource() {
        return favouritesLiveDataSource;
    }
}

File: app/src/main/java/com/example/unsplash/model/datasource/FavouritesDataSourceFactory.kt
package com.example.unsplash.model.datasource

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource

import com.example.unsplash.model.models.Photo

class FavouritesDataSourceFactory : DataSource.Factory<Int, Photo>() {

    private val favouritesLiveDataSource = MutableLiveData<PageKeyedDataSource<Int, Photo>>()

    override fun create(): DataSource<Int, Photo> {
        val favouritesDataSource = FavouritesDataSource()
        favouritesLiveDataSource.postValue(favouritesDataSource)
        return favouritesDataSource
    }
}